src/6_branch.js

```diff
@@ -1027,6 +1016,187 @@ Branch.prototype['sendSMS'] = wrap(
+ * | --- | ---
+ * | "$desktop_url" | Where to send the user on a desktop or laptop. By default it is the Branch-hosted text-me service
+ * | "$android_url" | The replacement URL for the Play Store to send the user if they don't have the app. _Only necessary if you want a mobile web splash_
+ * | "$ios_url" | The replacement URL for the App Store to send the user if they don't have the app. _Only necessary if you want a mobile web splash_
+ * | "$ipad_url" | Same as above but for iPad Store
+ * | "$fire_url" | Same as above but for Amazon Fire Store
+ * | "$blackberry_url" | Same as above but for Blackberry Store
+ * | "$windows_phone_url" | Same as above but for Windows Store
+ * | "$after_click_url" | When a user returns to the browser after going to the app, take them to this URL. _iOS only; Android coming soon_
+ *
+ * You have the ability to control the direct deep linking of each link as well:
+ *
+ * | Key | Value
+ * | --- | ---
+ * | "$deeplink_path" | The value of the deep link path that you'd like us to append to your URI. For example, you could specify "$deeplink_path": "radio/station/456" and we'll open the app with the URI "yourapp://radio/station/456?link_click_id=branch-identifier". This is primarily for supporting legacy deep linking infrastructure.
+ *
+ * #### Usage
+ * ```js
+ * branch.deepview(
+ *     data,
+ *     options,
+ *     callback (err)
+ * );
+ * ```
+ *
+ * #### Example
+ * ```js
+ * branch.deepview(
+ *     {
+ *         channel: 'facebook',
+ *         data: {
+ *             mydata: 'content of my data',
+ *             foo: 'bar',
+ *             '$desktop_url': 'https://en.wikipedia.org/wiki/Internet',
+ *             '$ios_url': 'https://en.wikipedia.org/wiki/Internet',
+ *             '$ipad_url': 'https://en.wikipedia.org/wiki/Internet',
+ *             '$android_url': 'https://en.wikipedia.org/wiki/Internet',
+ *             '$deepview_path': 'item_id=12345',
+ *             '$og_app_id': '12345',
+ *             '$og_title': 'My App',
+ *             '$og_description': 'My app\'s description.'
+ *         },
+ *         feature: 'dashboard',
+ *         stage: 'new user',
+ *         tags: [ 'tag1', 'tag2' ],
+ *     },
+ *     {
+ *         make_new_link: true,
+ *         open_app: true
+ *     },
+ *     function(err) {
+ *         console.log(err || 'no error');
+ *     }
+ * );
+ * ```
+ *
+ * ##### Callback Format
+ * ```js
+ * callback(
+ *     "Error message"
+ * );
+ * ```
+ *
+ */
+/*** +TOC_ITEM #deepviewdata-options-callback &.deepview()& ^ALL ***/
+Branch.prototype['deepview'] = wrap(
+	callback_params.CALLBACK_ERR,
+	function(done, data, options) {
+		var self = this;
+
+		if (!options) {
+			options = { };
+		}
+		var cleanedData = utils.cleanLinkData(data);
+
+		if (cleanedData['tags']) {
+			cleanedData['tags'] = goog.json.serialize(cleanedData['tags']);
+		}
+
+		if (cleanedData['data']) {
+			cleanedData['metadata'] = cleanedData['data'];
+			delete cleanedData['data'];
```

Comment:
app/index.js does this parsing thing for req.query.data for all incoming GET requests, which has some base64blahblah logic that I don't want to further complicate just for deepview's sake, and since we only expect users to call the deepview api endpoint directly, I figure I'd just pass in metadata, instead of data, to bypass the base64blahblah parsing (if we pass in data like this, it's gonna yell "unable to parse data").

In short, it was a lazy hack so that I can piggyback on utils.cleanLinkData() here without having to bother with the more complicated stringification/parsing mechanisms.